nemo/collections/llm/gpt/data/pre_training.py

```diff
@@ -207,27 +216,46 @@ def __init__(
             rampup_batch_size=rampup_batch_size,
         )
 
-    def setup(self, stage: str = "") -> None:
+    def build(
+        self,
+        trainer_max_steps: int,
+        trainer_val_check_interval: int,
+        trainer_limit_val_batches: Union[int, float],
+        trainer_limit_test_batches: Union[int, float],
+    ):
         from megatron.core.datasets.blended_megatron_dataset_builder import BlendedMegatronDatasetBuilder
         from megatron.core.datasets.gpt_dataset import GPTDataset
 
-        assert (
-            hasattr(self, "trainer") and self.trainer is not None
-        ), "Setup should be completed when trainer and config are attached."
+        train_iters = trainer_max_steps
+        assert train_iters > 0, f"max_steps {train_iters} should be greater than 0"
+        num_train_samples = int(train_iters * self.data_sampler.global_batch_size)
+
+        if self.num_train_samples is not None:
+            assert (
+                self.num_train_samples > num_train_samples
+            ), f"num_train_samples must be greater than {num_train_samples}."
```

Comment:
Should this be greater than or equal to?

Reply:
Addressed in https://github.com/NVIDIA/NeMo/pull/11088/commits/9494483197fa5ab915da2a2f8c1e9bf8075bba6a